paimon-core/src/main/java/org/apache/paimon/operation/AppendOnlyFileStoreWrite.java

```diff
@@ -137,6 +168,7 @@ protected RecordWriter<InternalRow> createWriter(
                         : new AppendOnlyCompactManager(
                                 compactExecutor,
                                 restoredFiles,
+                                dvIndexFileMaintainer,
```

Comment:
Only when skipCompaction is false, we can create `DeletionVectorIndexFileMaintainer`. And bucketMode must not be `BucketMode.BUCKET_UNAWARE`.

Reply:
I think so, but I can't confirm this bucket mode is not `BucketMode.BUCKET_UNAWARE`, so I still use the `bucketMode != BucketMode.BUCKET_UNAWARE` instead of `true`.